### PR TITLE
Added ability for a user to set their timezone.

### LINKED ISSUE #164: Add time zone to user profile
Currently, users can specify their location in their profile settings. A common use case is also for the user to be able to specify their timezone. We can then write a function to convert UTC timestamps to the user's local time. Are there plans for this on the roadmap (a search does not reveal any)? I would see this as: 

* Add timezone column in profile table
* Add timezone dropdown list in profile settings view
* Store selected timezone upon profile save
* Ideally add a function to call time for user; something like: Yii::$app->user->toLocalDateTime() with an optional argument that can be passed to convert a specified UTC time, defaulting to the current UTC time if no argument is passed

### PR DESCRIPTION
**Pull request by yarrgh**
Added ability for a user to set their timezone. Defaults to the Yii application timezone if not provided during registration.

Fixes #164

### CODE REVIEW

**Comment by dmeroff** [comment]
I'll review your changes on this weekend, sorry for the delay.

**Comment by yarrgh** [comment]
No worries. Let me know if anything should be changed or if you can find a better way of doing anything.

File: migrations/m151218_234654_add_timezone_to_profile.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+use yii\db\Schema;
+use yii\db\Migration;
+
+class m151218_234654_add_timezone_to_profile extends Migration
+{
+    public function up()
+    {
+        $this->addColumn('{{%profile}}', 'timezone', Schema::TYPE_STRING . (255));
```
Comment (dmeroff) [inline migrations/m151218_234654_add_timezone_to_profile.php]: This line should be like:
```php
$this->addColumn('{{%profile}}', 'timezone', Schema::TYPE_STRING . '(255)');
```

Comment (dmeroff) [inline migrations/m151218_234654_add_timezone_to_profile.php]: Moreover, do we really need 255 chars here? What is the longest possible timezone name?

File: migrations/m151218_234654_add_timezone_to_profile.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    public function down()
+    {
+        $this->dropcolumn('{{%profile}}', 'timezone');
+        return false;
```
Comment (dmeroff) [inline migrations/m151218_234654_add_timezone_to_profile.php]: This method should not return false.

File: models/Profile.php [hunk 3]
```diff
@@ -47,6 +56,7 @@ public function rules()
     {
         return [
             'bioString' => ['bio', 'string'],
+            'profileString' => ['timezone', 'string', 'max' => 255],
```
Comment (dmeroff) [inline models/Profile.php]: I think we need more complex rule here: we should check if the timezone really exist.

File: models/User.php [hunk 4]
```diff
@@ -241,13 +241,15 @@ public function create()
         $this->confirmed_at = time();
         $this->password = $this->password == null ? Password::generate(8) : $this->password;
 
+
+
```
Comment (dmeroff) [inline models/User.php]: This empty lines should be removed

File: models/User.php [hunk 5]
```diff
@@ -241,13 +241,15 @@ public function create()
...
         $this->trigger(self::BEFORE_CREATE);
 
         if (!$this->save()) {
             return false;
         }
 
-        $this->mailer->sendWelcomeMessage($this, null, true);
+        $this->mailer->sendWelcomeMessage($this, null, true);
```
Comment (dmeroff) [inline models/User.php]: This empty space at the end should be removed

File: helpers/Timezone.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+{
+
+    /**
+     * Get all of the time zones with the offsets sorted by their offset
+     *
+     * @return array
+     */
+    public static function getAll() {
+        $timezones = [];
+        $time_zones = $timezone_identifiers = \DateTimeZone::listIdentifiers();
```
Comment (dmeroff) [inline helpers/Timezone.php]: You should follow PSR-1/PSR-2, so var names should be written in camel case.

**Comment by yarrgh** [comment]
Just made the necessary changes. Not sure why the Travis CI test is failing. Not able to create a directory due to permission issues.

File: models/Profile.php [hunk 7]
```diff
@@ -94,14 +93,14 @@ public function attributeLabels()
...
-            'timezone'       => Yii::t('user', 'Time zone'),
+            'timezone'       => \Yii::t('user', 'Time zone'),
         ];
     }
     
     public function validateTimeZone($attribute, $params)
     {
         if (!in_array($this->$attribute, timezone_identifiers_list())) {
-            $this->addError($attribute, 'Time zone is not a valid time zone');
+            $this->addError($attribute, \Yii::t('user', 'Time zone is not valid');
```
Comment (dmeroff) [inline models/Profile.php]: This line is missing one more `)` right before semicolon

File: models/Profile.php [hunk 8]
```diff
@@ -94,14 +93,14 @@ public function attributeLabels()
             'location'       => \Yii::t('user', 'Location'),
             'website'        => \Yii::t('user', 'Website'),
             'bio'            => \Yii::t('user', 'Bio'),
-            'timezone'       => Yii::t('user', 'Time zone'),
+            'timezone'       => \Yii::t('user', 'Time zone'),
         ];
     }
     
     public function validateTimeZone($attribute, $params)
```
Comment (dmeroff) [inline models/Profile.php]: I think we need phpdoc here, which describes what this method does.

**Comment by dmeroff** [comment]
@yarrgh I've fixed travis build.

**Comment by dmeroff** [comment]
There are also a couple of feature which can be added:

- Add method to user profile model which returns DateTimeZone object for user's timezone
- Add method which can take `DateTime` object as a parameter, then set user's timezone and return modified object.

It will be super cool, if you could add these methods!

**Comment by yarrgh** [comment]
Still need to add the second suggested method. Will add it once I get a little more time.

**Comment by yarrgh** [comment]
Both suggested methods were added. What do you think?

**Comment by dmeroff** [comment]
Thank you! You did amazing job.

**Pull request merged**